networkx/algorithms/tests/test_d_separation.py

```diff
@@ -0,0 +1,79 @@
+from itertools import combinations
+import pytest
+import networkx as nx
+
+
+class TestDSeparation:
+    @classmethod
+    def setup_class(cls):
+        cls.path_G = nx.path_graph(3, create_using=nx.DiGraph)
+        cls.path_G.graph["name"] = "path"
+        nx.freeze(cls.path_G)
+
+        cls.fork_G = nx.DiGraph(name="fork")
+        cls.fork_G.add_edges_from([(0, 1), (0, 2)])
+        nx.freeze(cls.fork_G)
+
+        cls.collider_G = nx.DiGraph(name="collider")
+        cls.collider_G.add_edges_from([(0, 2), (1, 2)])
+        nx.freeze(cls.collider_G)
+
+        cls.naive_bayes_G = nx.DiGraph(name="naive_bayes")
+        cls.naive_bayes_G.add_edges_from([(0, 1), (0, 2), (0, 3), (0, 4)])
+        nx.freeze(cls.naive_bayes_G)
```

Comment:
This one was the hard one for me; the others I could catch. @arunwise, can you enlighten me here? 

`0` refers to the predictor (non-plural) vector `x`, and `1, 2, 3, 4` refer to the classes `y` - is this correct? I had always thought the NB model was the other way around - a whole bunch of predictors `1, 2, ... n` pointing to the class label output `0`.

Reply:
In the DAG of a NB model the root is the target variable and the features are its children. So in this example `0` is the node corresponding to target variable and `1,2,3,4` are the nodes corresponding to the features. The details like number of classes is not part of the graph. I hope that answers your question.

Reply:
Got it, thanks!